akka-serialization-jackson/src/main/scala/akka/serialization/jackson/JacksonSerializer.scala

```diff
@@ -0,0 +1,369 @@
+/*
+ * Copyright (C) 2016-2019 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.serialization.jackson
+
+import java.io.ByteArrayInputStream
+import java.io.ByteArrayOutputStream
+import java.io.NotSerializableException
+import java.util.zip.GZIPInputStream
+import java.util.zip.GZIPOutputStream
+
+import scala.annotation.tailrec
+import scala.util.Failure
+import scala.util.Success
+import scala.util.control.NonFatal
+
+import akka.actor.ExtendedActorSystem
+import akka.annotation.InternalApi
+import akka.event.LogMarker
+import akka.event.Logging
+import akka.serialization.BaseSerializer
+import akka.serialization.SerializationExtension
+import akka.serialization.SerializerWithStringManifest
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.databind.jsontype.impl.SubTypeValidator
+import com.fasterxml.jackson.dataformat.cbor.CBORFactory
+import com.fasterxml.jackson.dataformat.smile.SmileFactory
+
+/**
+ * INTERNAL API
+ */
+@InternalApi private[akka] object JacksonSerializer {
+
+  /**
+   * Using the blacklist from Jackson databind of class names that shouldn't be allowed.
+   * Not nice to depend on implementation details of Jackson, but good to use the same
+   * list to automatically have the list updated when new classes are added in Jackson.
+   */
+  class GadgetClassBlacklist extends SubTypeValidator {
```

Comment:
That implementation looks a bit magic. Are the details (Spring and C3P0) also taken from jackson? Auto-converted from Java even? Looks a bit weird ;)

Reply:
I included the Spring and C3P0 checks from Jackson SubTypeValidator because I found no good way to reuse them. The primary goal was to reuse the same blacklist `SubTypeValidator.DEFAULT_NO_DESER_CLASS_NAMES`.

Talking about weird code: https://github.com/FasterXML/jackson-databind/blob/master/src/main/java/com/fasterxml/jackson/databind/jsontype/impl/SubTypeValidator.java#L106-L138
It was not auto-converted, but maybe something weird was passed along. Please suggest what can be better.